{{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/settings.py

```diff
@@ -118,10 +143,24 @@
 
 MEDIA_ROOT = env('MEDIA_ROOT', default=root('media'))
 
-{% if cookiecutter.use_https == "y" %}
-# To make request.scheme as https when request comes through nginx
-SECURE_PROXY_SSL_HEADER = ('HTTP_X_SCHEME', 'https')
-{% endif %}
+# redirect HTTP to HTTPS
+if env.bool('HTTPS_REDIRECT', default=False) and not DEBUG:
+    SECURE_SSL_REDIRECT = True
+    SECURE_REDIRECT_EXEMPT = []
+    SESSION_COOKIE_SECURE = True
+    CSRF_COOKIE_SECURE = True
+else:
+    SECURE_SSL_REDIRECT = False
+
+# trust the given (by default "X-Scheme") header that comes from our proxy (nginx),
+# and any time its value is "https",
+# then the request is guaranteed to be secure (i.e., it originally came in via HTTPS).
+HTTPS_PROXY_HEADER = env('HTTPS_PROXY_HEADER').upper()
```

Comment:
please test to see if nginx deletes this header if the mitm attacker provides it and passes the request through http

Reply:
How can someone modify our environment variable? Also, I think this should not be an env variable but hardcoded in code because it's hardcoded in nginx config.

Reply:
Not environment variable. Header. Someone can send any header.

You are right regarding hardcoding. It still needs to be checked though, for security.

Reply:
How can I test HTTPS on my local machine?

Reply:
Could you use moa?

Reply:
I tested on moa and it is fine.